src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -563,24 +554,13 @@ public virtual void WriteLiteralTo([NotNull] TextWriter writer, string value)
                     // Otherwise the value is another object (perhaps an HtmlString) and we'll ask it to format itself.
                     string stringValue;
 
-                    // Intentionally using is+cast here for performance reasons. This is more performant than as+bool?
-                    // because of boxing.
-                    if (val.Value is bool)
+                    if (!ShouldRenderAttributeValue(val))
```

Comment:
Incorrect since `value` is of type `PositionTagged<object>` and `ShouldRenderAttributeValue()` never checks its `Value`. Either change the `value` initialization to be `attributeValue.Value.Value` or restore use of `value.Value` in the rest of this method.